eoxserver/services/ows/wps/processes/get_statistics.py

```diff
@@ -0,0 +1,142 @@
+        DEM implementtation """
+
+    def __init__(self):
+        super(GetStatisticsProcess, self).__init__()
+
+    implements(ProcessInterface)
+
+    identifier = "GetStatistics"
+    title = "Get statistics for a coverage/s that intersects with the input bbox"
+    description = ("provides statistics of all the coverages whithin a provided bounding box. "
+                  " The process is used by the  by the EOxC DEM implementtation")
+    metadata = {}
+    profiles = ['EOxServer:GetStatistics']
+
+    inputs = {
+        "bbox": LiteralData(
+            "bbox",
+            title="bounding box that intersect with the products."
+        ),
+    }
+
+    outputs = {
+        "statistic": ComplexData(
+            "statistic",
+            title="output statistics",
+            abstract="coverage/s statistics in json format.",
+            formats=FormatJSON()
+        ),
+    }
+
+    @staticmethod
+    def execute(bbox, **kwarg):
+        """ The main execution function for the process.
+        """
+
+        values = list(map(float, bbox.split(",")))
+        if len(values) != 4:
+            raise ValueError("Invalid number of coordinates in 'bbox'.")
+
+        parsed_bbox = Polygon.from_bbox(values)
+
+        # get the dataset series intersecting with the requested bbox
+
+        coverages = models.Coverage.objects.filter(
+            Q(footprint__intersects=parsed_bbox)
+            | Q(footprint__isnull=True, parent_product__footprint__intersects=parsed_bbox))
+
+        report = {}
+        for index, coverage in enumerate(coverages):
+
+            try:
+                data_items = coverage.arraydata_items.all()
+
+            except coverage.arraydata_items.all().length > 1:
+                raise InvalidInputValueError(
+                    "coverage", "coverage '%s' has more than one imagery, this process handles single images!" % coverage
+                )
+
+            data_item = data_items[0]
+
+            ds = gdal_open(data_item, False)
+
+            band = ds.GetRasterBand(1)
+            image_array = band.ReadAsArray()
+            stats = ds.GetRasterBand(1).GetStatistics(0, 1)
+            bin_array, hist = np.histogram(image_array, bins=np.arange(int(stats[0]), int(stats[1]), 20, int))
+
+            # nodata = band.GetNoDataValue()
+            # no_value = np.where(np.any(image_array==nodata, axis=1))
+
+            id_key = "Coverage_%s" % str(index + 1)
+            coverage_id = coverage.identifier
+
+            stats_json = {
+                "id": coverage_id,
+                "MINIMUM": stats[0],
+                "MAXIMUM": stats[1],
+                "MEAN": stats[2],
+                "STDDEV": stats[3],
+                "HISTOGRAM_FREQUENCY": bin_array.tolist(),
+                "HISTOGRAM_PIXEL_VALUES": hist.tolist(),
+                # "NUMBER_OF_NODATA_PIXELS": no_value[0].size
+                }
+            report[id_key] = stats_json
```

Comment:
It is not quite clear why you need an `id_key` here. The `coverage_id` would be sufficient.

Also: either use a JSON object with the coverage ID as keys or (which I think is preferrable) simply use a JSON array with the stats object you create.